python-modules/cis_change_service/cis_change_service/profile.py

```diff
@@ -0,0 +1,97 @@
+"""Get the status of an integration from the identity vault and auth0."""
+import json
+from cis_aws import connect
+
+
+class Vault(object):
+    """Handles flushing profiles to Dynamo when running local or in stream bypass mode."""
+    def __init__(self, sequence_number):
+        self.connection_object = connect.AWS()
+        self.identity_vault_client = None
+        self.sequence_number = sequence_number
+
+    def _connect(self):
+        self.connection_object.session()
+        self.connection_object.assume_role()
+        self.identity_vault_client = self.connection_object.identity_vault_client()
+        return self.identity_vault_client
+
+    def put_profile(self, profile_json):
+        """Write profile to the identity vault."""
+        if self.identity_vault_client is None:
+            self._connect()
+
+        user_id = self._get_id(profile_json)
+        primary_email = self._get_primary_email(profile_json)
+        dynamodb_schema_dict = {
+            'profile': {'S': json.dumps(profile_json)},
+            'id': {'S': user_id},
+            'sequence_number': {'S': self.sequence_number},
+            'primaryEmail': {'S': primary_email}
```

Comment:
i wonder if you want to change this with profilev2 (it doesnt have `primaryEmail` as a field) (i know this means a dynamodb change)

Reply:
There is a value I can pull out that is the users email... that's what is happening at present.

Reply:
What should I put here instead? `'primaryEmail': {'S': primary_email}`

Reply:
We have to store primary email as a top level attribute separate from the map because it needs to be searchable in a global secondary index.  I may be misinterpreting this... looking at the profile.  Do you mean s/primaryEmail/primary_email/g ?

Reply:
If so this is an easy change.  I also need to update the schema to add a partiton key to this.

Reply:
yes mainly s/primaryEmail/primary_email/ so it reflects the profile data field name